app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java

```diff
@@ -128,8 +136,10 @@ public void setPlaybackQuality(@Nullable final String playbackQuality) {
     }
 
     public interface QualityResolver {
-        int getDefaultResolutionIndex(List<VideoStream> sortedVideos);
+        int getDefaultResolutionIndex(StreamInfo streamInfo, List<VideoStream> sortedVideos);
```

Comment:
Mmmh, by looking at the code above it seems like `StreamInfo` will be of livestream type only when `maybeBuildLiveMediaSource` fails (which should not happen for livestreams, unless `getHlsUrl` or `getDashMpdUrl` return empty urls) (`VideoPlaybackResolver` is the only place where `getDefaultResolutionIndex` and `getOverrideResolutionIndex` are *used*). So having `if (StreamTypeUtil.isLiveStream(streamInfo.getStreamType())) { return -1; }` in the player above is useless, I think (that's the only place where `getDefaultResolutionIndex` and `getOverrideResolutionIndex` are *overridden*).

Reply:
Not sure about that, I'd like to hear advice of other team members.

Reply:
Well, let's keep it and just add a comment explaining that `return -1` should be unreachable

Reply:
> So having `if (StreamTypeUtil.isLiveStream(streamInfo.getStreamType())) { return -1; }` in the player above is useless

Indeed, I just look better at the code and you are right, that's useless to set `-1` and this can even create bugs because no video source would be available for livestreams (because `MediaSourceTag.getSelectedVideoStream()` will return `null`).

So I will rollback my changes here and keep code as it was before.